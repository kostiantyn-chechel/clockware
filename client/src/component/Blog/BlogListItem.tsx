import React from 'react';
import makeStyles from "@material-ui/core/styles/makeStyles";
import {PostAttributes} from "../../../../models/post.model";
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import CardActions from '@material-ui/core/CardActions';
import IconButton from '@material-ui/core/IconButton';
import EditIcon from '@material-ui/icons/Edit';
import DeleteForeverIcon from '@material-ui/icons/DeleteForever';
import {dayToString} from "../../helpers/dateTime";

const DEFAULT_PHOTO = 'https://res.cloudinary.com/kodevtm/image/upload/v1612537822/clockware/ao4a0dne0siugaud9pbm.jpg';

const useStyles = makeStyles((theme) => ({
    card: {
        marginTop: theme.spacing(3),
        width: '80%',
        minHeight: '200px',
        minWidth: '460px',
        maxWidth: '800px',
    },
    media: {
        height: 140,
    },
    buttons: {
        display: 'flex',
        justifyContent: 'space-between'
    }
}));

interface IBlogListItem {
    post: PostAttributes
    isEdit: boolean
    handleDelPost: (id: number) => void
    handleEditPost: (id: number) => void
}

const BlogListItem: React.FC<IBlogListItem> = (props) => {
    const classes = useStyles();
    const { post, isEdit, handleDelPost, handleEditPost } = props;

    return (
        <Card className={classes.card}>
            <CardMedia
                className={classes.media}
                image={post.photoURL || DEFAULT_PHOTO}
                title={`Clockware post #${post.id || 0}`}
            />
            <CardContent>
                <div className={classes.buttons}>
                    <div>#{post.id}</div>
                    <div>{dayToString(post.date)}</div>
                </div>

                <div dangerouslySetInnerHTML={{__html: post.post}} />
            </CardContent>

            {isEdit ?
                <CardActions className={classes.buttons} disableSpacing>
                    <IconButton
                        aria-label="edit"
                        onClick={() => handleEditPost(post.id!)}
                    >
                        <EditIcon />
                    </IconButton>
                    <IconButton
                        aria-label="delete"
                        onClick={() => handleDelPost(post.id!)}
                    >
                        <DeleteForeverIcon />
                    </IconButton>
                </CardActions>
                : null
            }

        </Card>
    );
};

export default BlogListItem;