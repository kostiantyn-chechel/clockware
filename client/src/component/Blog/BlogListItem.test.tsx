import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import CardActions from '@material-ui/core/CardActions';
configure({ adapter: new Adapter() });

import BlogListItem from './BlogListItem';

describe('Test BlogListItem component', () => {
    let isEdit = false;
    const handleDelPostMock = jest.fn();
    const handleEditPostMock = jest.fn();
    const post = {
        id: 33,
        date: '06-01-2021',
        post: 'some text',
        photoURL: 'https://res.cloudinary.com/iugaud9pbm.jpg'
    };

    const Component = <BlogListItem
        post={post}
        isEdit={isEdit}
        handleDelPost={handleDelPostMock}
        handleEditPost={handleEditPostMock}
    />;
    const ShallowBlogListItem = shallow(Component);
    
    test('BlogListItem component should exists', () => {
        expect(ShallowBlogListItem).toBeDefined();
    });

    test("ToMatchSnapshot BlogListItem component", () => {
        expect(renderer.create(Component).toJSON()).toMatchSnapshot();
    });

    //TODO ????
    test("BlogListItem enabled Control Panel (Edit and Del button)", () => {
        const Component = <BlogListItem
            isEdit={true}
            post={post} handleDelPost={handleDelPostMock} handleEditPost={handleEditPostMock}
        />;
        const MountBlogListIte = mount(Component);

        const BlogListItemComponent = MountBlogListIte.find(CardActions);
        expect(BlogListItemComponent).toBeDefined(); //TODO ????

    });

    //TODO ????
    test("BlogListItem disabled Control Panel (Edit and Del button)", () => {
        const Component = <BlogListItem
            isEdit={false}
            post={post} handleDelPost={handleDelPostMock} handleEditPost={handleEditPostMock}
        />;
        const MountBlogListItem = mount(Component);

        expect(MountBlogListItem.contains(<CardActions />)).toBeDefined(); //TODO ????

    });

});