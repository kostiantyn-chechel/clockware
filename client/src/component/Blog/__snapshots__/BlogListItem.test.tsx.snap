// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BlogListItem component ToMatchSnapshot BlogListItem component 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardMedia-root makeStyles-media-2"
    style={
      Object {
        "backgroundImage": "url(\\"https://res.cloudinary.com/iugaud9pbm.jpg\\")",
      }
    }
    title="Clockware post #33"
  />
  <div
    className="MuiCardContent-root"
  >
    <div
      className="makeStyles-buttons-3"
    >
      <div>
        #
        33
      </div>
      <div>
        06-01-2021
      </div>
    </div>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "some text",
        }
      }
    />
  </div>
</div>
`;
